packages/list-store/src/search-params.ts

```diff
@@ -11,14 +13,41 @@ const searchParamFilterSchema = z
   .or(z.number().transform(value => `${value}`))
```

Comment:
Unrelated to this PR - this just comes to mind: I remember that the new version of Zod has improved its `coerce` implementation [1]. Can we use this to simplify the code?

[1] https://zod.dev/?id=coercion-for-primitives

Reply:
Nice, we can definitely use this